src/coreclr/vm/jitinterface.cpp

```diff
@@ -11788,6 +11788,47 @@ void* CEEJitInfo::getFieldAddress(CORINFO_FIELD_HANDLE fieldHnd,
     return result;
 }
 
+void* CEEJitInfo::getFrozenHandleFromInitedStaticField(CORINFO_FIELD_HANDLE fieldHnd)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    void* ptr = nullptr;
+
+    JIT_TO_EE_TRANSITION();
+
+    FieldDesc* field = (FieldDesc*)fieldHnd;
+    if (field->IsStatic() && field->IsObjRef() && !field->IsThreadStatic())
+    {
+        MethodTable* pEnclosingMT = field->GetEnclosingMethodTable();
+        if (!pEnclosingMT->IsSharedByGenericInstantiations())
+        {
+            // Allocate space for the local class if necessary, but don't trigger
+            // class construction.
+            DomainLocalModule* pLocalModule = pEnclosingMT->GetDomainLocalModule();
+            pLocalModule->PopulateClass(pEnclosingMT);
+
+            GCX_COOP();
+            OBJECTREF fieldObj = field->GetStaticOBJECTREF();
+            if (fieldObj != NULL && pEnclosingMT->IsClassInited() && IsFdInitOnly(field->GetAttributes()))
```

Comment:
We bake-in null values null. (It would require changing the method signature to return bool + value.)

Reply:
Do you mean unrelated to frozen objects if we know that the value is null we can import it as null? e.g.:
```csharp
private static string MyConfig = Environment.GetEnvironmentVariable("NonExistingEnvVar");
```
? (`GetEnvironmentVariable` returns null if env var is not defined)

Reply:
Right. null ref is kind of a special frozen object ref.

Reply:
Actually, it would be nice to generalize this method signature for both primitive and objectref known values and switch `impImportStaticReadOnlyField` to it. It would allow `impImportStaticReadOnlyField` to work for aot preinitialization too.